plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/ExchangeS3Config.java

```diff
@@ -270,4 +278,37 @@ public ExchangeS3Config setGcsJsonKey(String gcsJsonKey)
         this.gcsJsonKey = Optional.ofNullable(gcsJsonKey);
         return this;
     }
+
+    @NotNull
+    public S3SseType getSseType()
+    {
+        return sseType;
+    }
+
+    @Config("exchange.s3.sse.type")
+    @ConfigDescription("Type of S3 server-side encryption for FTE")
+    public ExchangeS3Config setSseType(S3SseType sseType)
+    {
+        this.sseType = sseType;
+        return this;
+    }
+
+    public Optional<String> getSseKmsKeyId()
+    {
+        return sseKmsKeyId;
+    }
+
+    @Config("exchange.s3.sse.kms-key-id")
+    @ConfigDescription("KMS Key ID to use for S3 server-side encryption with KMS-managed key for FTE")
+    public ExchangeS3Config setSseKmsKeyId(String sseKmsKeyId)
+    {
+        this.sseKmsKeyId = Optional.ofNullable(sseKmsKeyId);
+        return this;
+    }
+
+    @AssertTrue(message = "Property value of exchange.s3.sse.kms-key-id needs to be provided only when exchange.s3.sse.type=KMS")
+    public boolean isSseConfigValid()
+    {
+        return sseType == KMS ^ getSseKmsKeyId().isEmpty();
+    }
```

Comment:
nit: Would be a bit more readable to have two separate validations for
* sseType != KMS && sskeKeyId not emtpy
* sseType == KMS && sskKeyId empty

Then you could provide easier to understand error messages.

Reply:
done